### PR TITLE
[Form] deprecated "property_path" in favor of "map_to"

### PR DESCRIPTION
**Pull request by sdecandelario**
[Form] deprecated "property_path" in favor of "map_to"

| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| Fixed tickets | #7832 |
| License | MIT |
| Doc PR | no |

### CODE REVIEW

File: src/Symfony/Component/Form/Tests/Extension/Core/Type/FormTypeTest.php [hunk 1]
```diff
@@ -529,7 +529,7 @@ public function testOverrideErrorBubbling()
     public function testPropertyPath()
     {
         $form = $this->factory->create('form', null, array(
-            'property_path' => 'foo',
+            'map_to' => 'foo',
```
Comment (webmozart) [inline src/Symfony/Component/Form/Tests/Extension/Core/Type/FormTypeTest.php]: Can you please keep the old tests and prefix them with `testLegacy`?

File: src/Symfony/Component/Form/Tests/Extension/Core/EventListener/ResizeFormListenerTest.php [hunk 2]
```diff
@@ -113,7 +113,7 @@ public function testPreSubmitResizesUpIfAllowAdd()
 
         $this->factory->expects($this->once())
             ->method('createNamed')
-            ->with(1, 'text', null, array('property_path' => '[1]', 'attr' => array('maxlength' => 10), 'auto_initialize' => false))
+            ->with(1, 'text', null, array('map_to' => '[1]', 'max_length' => 10, 'auto_initialize' => false))
```
Comment (webmozart) [inline src/Symfony/Component/Form/Tests/Extension/Core/EventListener/ResizeFormListenerTest.php]: Did you remove the `attr` option on purpose? Please keep it.

File: src/Symfony/Component/Form/FormConfigBuilderInterface.php [hunk 3]
```diff
@@ -161,6 +161,17 @@ public function setRequired($required);
     public function setPropertyPath($propertyPath);
```
Comment (webmozart) [inline src/Symfony/Component/Form/FormConfigBuilderInterface.php]: I think the `setPropertyPath()` method should be deprecated. You also need to add the corresponding `trigger_error()` statements.

File: src/Symfony/Component/Form/Extension/Core/Type/FormType.php [hunk 4]
```diff
@@ -194,6 +194,7 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
             'read_only'          => false,
             'max_length'         => null,
             'pattern'            => null,
+            'map_to'             => null,
```
Comment (webmozart) [inline src/Symfony/Component/Form/Extension/Core/Type/FormType.php]: The "map_to" option should receive the value of the "property_path" option by default. See how the options "inherit_data"/"virtual" are handled in FormType.

You should also add a normalizer for the "property_path" option that triggers a deprecation error if the option is set. See how "choice_list" is handled in ChoiceType for that.

File: src/Symfony/Component/Form/FormConfigBuilderInterface.php [hunk 5]
```diff
@@ -161,6 +161,17 @@ public function setRequired($required);
...
     /**
+     * Sets the property path that the form should be mapped to.
+     *
+     * @param null|string|PropertyPathInterface $propertyPath
+     *             The property path or null if the path should be set
+     *             automatically based on the form's name.
+     *
+     * @return self The configuration object.
+     */
+    public function setMapTo($propertyPath);
```
Comment (webmozart) [inline src/Symfony/Component/Form/FormConfigBuilderInterface.php]: This needs to be documented in the UPGRADE file. See our BC policy on [adding methods to non-API interfaces](http://symfony.com/doc/current/contributing/code/bc.html#changing-interfaces).

**Comment by webmozart** [comment]
This looks good in general, thanks @hades87! Can you do the changes I mentioned and rebase this on 2.8?

File: src/Symfony/Component/Form/Tests/Extension/Core/Type/FormTypeTest.php (hunk 1 above)
Comment (xabbuh) [inline src/Symfony/Component/Form/Tests/Extension/Core/Type/FormTypeTest.php]: There's no need to prefix them. Adding them to the `legacy` group is enough:

```php
/**
 *  @group legacy
 */
public function testPropertyPath()
{
   // ...
}
```

Comment (webmozart) [inline src/Symfony/Component/Form/Tests/Extension/Core/Type/FormTypeTest.php]: You are right. I didn't consider that the old test methods are called `testPropertyPathXxx` while the new  ones would be called `testMapToXxx`.

**Comment by Tobion** [comment]
Is it really worth to change that now? The ticket says it would allow other mappers like XML. But I feel it only makes sense to change the naming when we actually have multiple mappers. No need to create BC breaks and maintainance for no real use-case yet.

**Comment by fabpot** [comment]
I tend to agree with @Tobion.

**Comment by webmozart** [comment]
Fair enough! :)

**Pull request closed without merging**